pkg/controller/controller_utils.go

```diff
@@ -942,16 +942,6 @@ func PatchNodeTaints(c clientset.Interface, nodeName string, oldNode *v1.Node, n
 		return fmt.Errorf("failed to marshal old node %#v for node %q: %v", oldNode, nodeName, err)
 	}
 
-	newTaints := newNode.Spec.Taints
```

Comment:
In general we shouldn't modify object that we are getting as an argument. And I think this change is incorrect.